### PR TITLE
Allow new `manifest.workspaces.packages` structure

### PR DESCRIPTION
**Pull request by klaasman**
As introduced here: https://yarnpkg.com/blog/2018/02/15/nohoist/

Workspaces can now be formatted as:
```js
{
  "workspaces": {
    "packages": [/* ... */],
    "nohoist": [/* ... */]
  }
}
```

The old workspaces format will continue to be supported.

### CODE REVIEW

**Comment by eventualbuddha** [comment]
Thanks! We should add tests to this before merging.

**Comment by klaasman** [comment]
It actually caught an unforeseen error 👍

File: index.js [hunk 1]
```diff
@@ -34,6 +36,11 @@ function findWorkspaceRoot(initial) {
   return null;
 }
 
+function extractWorkspaces(manifest) {
+  const workspaces = (manifest || {}).workspaces;
+  return (workspaces && workspaces.packages) || (workspaces instanceof Array ? workspaces : null);
```
Comment (eventualbuddha) [inline index.js]: Should prefer `Array.isArray(workspaces)`.

**Review by eventualbuddha — CHANGES_REQUESTED** [review]
Thanks for adding tests. Looks like there’s just one minor thing to fix before merging.

File: index.js (hunk 1 above)
Comment (klaasman) [reply index.js]: I guess this will break the CI Node.js 4 task too

Comment (eventualbuddha) [reply index.js]: It shouldn't…

```
> process.version
'v4.8.7'
> Array.isArray
[Function: isArray]
```

**Review by eventualbuddha — APPROVED** [review]

**Comment by eventualbuddha** [comment]
@thetimothyp @lennyburdette want to take it from here?

**Comment by lennyburdette** [comment]
@klaasman Thank you!

**Pull request merged**